proposals/2045-column-lineage-endpoint.md

```diff
@@ -66,78 +74,116 @@ In the GET /lineage api, add column lineage to DATASET nodes' data
 }
 ```
 
-### add a column-level-lineage endpoint:
+The implementation here can reuse `columnLineage` facet classes.
+
+
+### Add a column-level-lineage endpoint:
 
+New endpoints to retrieve a column lineage of a single field or a whole dataset will be added: 
 ```
-GET /column-lineage?nodeId=dataset:food_delivery:public.delivery_7_days&column=a
+GET /column-lineage?nodeId=dataset:{namespace}:{dataset}
+GET /column-lineage?nodeId=datasetField:{namespace}:{dataset}:{field}
+```
+For example:
+```
+GET /column-lineage?nodeId=dataset:food_delivery:public.delivery_7_days
+GET /column-lineage?nodeId=datasetField:food_delivery:public.delivery_7_days:a
 ```
-`column` is a ne parameter that must be a column in the schema of the provided dataset `nodeId`.
 
-The logic is layered on the existing lineage endpoint, filtering down to the datasets that contribute to that column.
-It only returns dataset nodes.
+Although creating a new endpoint, we would like to reuse existing data structures with a new `NodeType.FIELD` introduced.
 
-```diff
+The logic returns dataset field node:
+
+```
+GET /column-lineage?nodeId=datasetField:db1:table1:a
+...
 {
   graph: [
     {
-       "id": "dataset:db1:table2",
-       "type": "DATASET",
-       data: {
-         namespace: "DB1",
-         name: "table2",
->       columnLineage: {
->           "a": {
->              inputFields: [
->                {namespace: "DB1", name: "table1, "field": "a"}
->              ],
->              transformationDescription: "identical",
->              transformationType: "IDENTITY"
->           },
->          "b": ... other output fields
->        }
+       "id": "datasetField:db1:table1:a",
+       "type": "DATASET_FIELD",
+       "data": {
+         "namespace": "DB1",
+         "name": "table2",
+         "field": "a",
+         "type": "integer",
+         "transformationDescription": "identical",
+         "transformationType": "IDENTITY",    
+         "inputFields": [
+            { "namespace": "DBA", "name": "tableA", "field": "columnA"},
+            { "namespace": "DBB", "name": "tableB", "field": "columnB"},
+            { "namespace": "DBC", "name": "tableC", "field": "columnC"}
+         ]
+         "inEdges": [
+          {
+             "origin": "datasetField:db1:table1:a",
+             "destination": "datasetField:DBA:tableA:columnA"
+          },
+          {
+             "origin": "datasetField:db1:table1:a",
+             "destination": "datasetField:DBB:tableB:columnB"
+          },
+          {
+             "origin": "datasetField:db1:table1:a",
+             "destination": "datasetField:DBB:tableB:columnC"
+          }
+         ],
```

Comment:
This is redundant with inputFields. namespace and name in inputfields are always matching origin in inEdges. Maybe we combine those?

Reply:
I considered it may be useful to present inputFields on WEB UI. If so, it is beneficial to have this information redundant to avoid parsing edges to inputFields.